File: SumOfDigitOfANumberUsingRecursion.java
package com.recursion.examples;

public class SumOfDigitOfANumberUsingRecursion {

    public static int getSumOfDigitOfANumberUsingRecursion(int number){
        System.out.println("getSumOfDigitOfANumberUsingRecursion");
        System.out.println("Number : "+number);
        if (number ==0){
            return 0;
        }
        int a = number%10;
        int b = number/10;
        System.out.println("a : "+a);
        System.out.println("b : "+b);
        return a+getSumOfDigitOfANumberUsingRecursion(b);
    }
}

File: SumOfDigitOfANumberUsingRecursion.py
def SumOfDigitOfANumberUsingRecursion(number):
    if number == 0:
        return 0
    a = int(number%10)
    b = int(number/10)
    return a+SumOfDigitOfANumberUsingRecursion(b)

if __name__ == '__main__':
    input1 = 12345
    input2 = 234
    print(SumOfDigitOfANumberUsingRecursion(input1))
    print(SumOfDigitOfANumberUsingRecursion(input2))